packages/components/src/chart/d3chart/utils/bar-chart.js

```diff
@@ -68,7 +68,16 @@ export const drawBars = ( node, data, params, scales, formats, tooltip ) => {
 		.attr( 'pointer-events', 'none' )
 		.attr( 'tabindex', '0' )
 		.attr( 'aria-label', d => {
-			const label = params.mode === 'time-comparison' && d.label ? d.label : d.key;
+			let label = d.key;
+			if ( params.mode === 'time-comparison' ) {
+				if ( d.label ) {
+					label = d.label;
+				} else {
+					const dayData = data.find( e => e.date === d.date );
+					const date = tooltip.labelFormat( moment( dayData[ d.key ].labelDate ).toDate() );
+					label = params.interval === 'hour' ? `${ d.key } ${ date }` : date;
```

Comment:
I think this is a nice touch.  One small oddity is that today vs yesterday is read as "Today" and "Previous Period."  I would think this would be either Today/Yesterday or Current Period/Previous Period.

We might also want to add something in for quarterly reports while we're here since the currently quarterly just reads the starting month and I imagine that may be confusing to users.

Reply:
Right, those labels are the same as the ones used in the legend. I updated the logic behind them to make them more specific (_Yesterday_, _Previous week_, etc.), instead of displaying _Previous Period_ for all of them.

![image](https://user-images.githubusercontent.com/3616980/52959726-40b20e80-3397-11e9-97aa-87e5e8d377ac.png)

Let me know what do you think.